services/movie_session.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    sessions_queryset = MovieSession.objects.all()
+
+    if not session_date:
+        return sessions_queryset
+    session_date = datetime.strptime(session_date, "%Y-%m-%d")
+    return sessions_queryset.filter(show_time__date=session_date)
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    movie_session_to_update = MovieSession.objects.filter(id=session_id)
+
+    if show_time:
+        movie_session_to_update.update(show_time=show_time)
+    if movie_id:
+        movie_session_to_update.update(movie_id=movie_id)
+    if cinema_hall_id:
+        movie_session_to_update.update(cinema_hall_id=cinema_hall_id)
```

Comment:
Using `.filter().update()` is less efficient and can lead to silent failures if the session ID does not exist. Consider using `.get()` to retrieve the `MovieSession` instance and then update its fields directly. This will also allow you to handle exceptions more effectively.